src/components/AlgoliaAutocomplete/AlgoliaAutocomplete.js

```diff
@@ -1,10 +1,41 @@
 import React, { useRef, useEffect, createElement, Fragment } from 'react';
 import { render } from 'react-dom';
 import { navigate } from 'gatsby';
-import { autocomplete } from '@algolia/autocomplete-js';
+import { autocomplete, getAlgoliaResults } from '@algolia/autocomplete-js';
+import algoliasearch from 'algoliasearch/lite';
+
+import SearchResult from './SearchResult';
 import destroySearchOverlay from './destroySearchOverlay';
 
-const AlgoliaAutocomplete = (props) => {
+const searchClient = algoliasearch(
+  process.env.GATSBY_ALGOLIA_APP_ID,
+  process.env.GATSBY_ALGOLIA_SEARCH_KEY
+);
+
+const getSources = ({ query }) => {
+  return [{
+    sourceId: 'docs',
+    getItemUrl({ item }) {
+      return item.slug;
+    },
+    getItems() {
+      return getAlgoliaResults({
+        searchClient,
+        queries: [{
+          indexName: 'docs',
+          query,
+        }],
+      });
+    },
+    templates: {
+      item({ item, components }) {
+        return <SearchResult hit={item} components={components} />;
+      }
+    }
+  }];
+};
+
```

Comment:
This code is mostly just extracted out of `doc/Sidebar.js`. It's now closer to the place where it's used.